### ISSUE TITLE
Undefined symbol "HTTP_GET"

### ISSUE THREAD

**Issue by ynhhoJ** [Open-GTO/sa-mp-fixes#39]
После обновления fixes.inc на данную версию: https://github.com/Open-GTO/sa-mp-fixes/commit/6fa65b788695f2b47d2be5729fd1bfe9c128fb23
При попытке скомпилить мод выдаёт такую ошибку:
```pawno\include\rustext.inc(97) : error 017: undefined symbol "HTTP_GET"```
Если убрать fixes.inc , всё компилируется нормально.

**Comment by Y-Less** [comment]
Did you include a_http?

**Comment by ziggi** [comment]
I don't know why this happens, because [rustext.inc](https://github.com/ziggi/rustext) has these lines:
```Pawn
#if !defined HTTP
	#tryinclude <a_http>
#endif

#if !defined HTTP
	#tryinclude "a_http"
#endif

#if !defined HTTP
	#tryinclude "../a_http"
#endif
```

**Comment by ziggi** [comment]
Upd: everything works fine without [these lines](https://github.com/Open-GTO/sa-mp-fixes/blob/master/fixes.inc#L3052-L3057).

So, I think what fixes.inc should include a_http by default or this lines should be disabled if a_http is not included.

**Comment by Y-Less** [comment]
You are better checking for the include symbol:

```pawn
#if !defined _inc_a_http
	#tryinclude <a_http>
#endif

#if !defined _inc_a_http
	#tryinclude "a_http"
#endif

#if !defined _inc_a_http
	#tryinclude "../a_http"
#endif
```

Unless you are using the new compiler, in which case maybe check for some really unique symbol like `HTTP_ERROR_MALFORMED_RESPONSE`.

**Comment by Y-Less** [comment]
Or the general advice, given how much fixes.inc changes and how low-level it is, is to include it only after default includes.  So it should be:

```pawn
#include <a_samp>
#include <a_http>
// Redefinitions such as `MAX_PLAYERS` go here.
#include <fixes>
// Now all other includes.
#include <rustext>
```

**Comment by ziggi** [comment]
But it's still fixes.inc issue, because fixes.inc thinks what a_http defined by default and this is wrong.
My solution for this:
```Pawn
	#if defined HTTP
		#if defined _ALS_HTTP
			#error _ALS_HTTP defined
		#endif
		native CST_HTTP(index, type, const url[], const data[], const callback[]) = HTTP;
		#define _ALS_HTTP
		#define HTTP( CST_HTTP(
	#endif
```

**UPD:** OK, now I understand the problem. So, we can just include a_http by default.

**Comment by rt-2** [comment]
I also got this error from FCNPC's script when compiling one of my filterscript after updating ``fixes.inc``.

**Comment by Y-Less** [comment]
I changed the code to this:

```pawn
	// a_http
	// Try very hard to include it!  The first one should be enough.  Shame that
	// include has no include guard in it, making it harder to use with Zeex's
	// compiler.
	#if !defined HTTP_ERROR_MALFORMED_RESPONSE
		#tryinclude <a_http>
	#endif
	#if !defined HTTP_ERROR_MALFORMED_RESPONSE
		#tryinclude "a_http"
	#endif
	#if !defined HTTP_ERROR_MALFORMED_RESPONSE
		#tryinclude "..\a_http"
	#endif
	#if defined HTTP_ERROR_MALFORMED_RESPONSE
		#if defined _ALS_HTTP
			#error _ALS_HTTP defined
		#endif
		native CST_HTTP(index, type, const url[], const data[], const callback[]) = HTTP;
		#define _ALS_HTTP
		#define HTTP( CST_HTTP(
		// Because a_http can be included multiple times (on one compiler),
		// ensure that multiple definitions don't break (too quickly - there's
		// nothing we can do if it gets included a load of times, but that's an
		// issue you would get with the new compiler anyway).
		#define CST_HTTP(%0,%1,%2[],%3[],%4[]); a_http_included_too_many_times(%0,%1,%2[],%3[],%4[]) = HTTP;
	#endif
```

I tested against rustext.inc and it works now, not tried with FCNPC.  It will break eventually using the new compiler (i.e. if you include a_http about 3 times, because there's no include guard by default), but it does that anyway without fixes.inc, so that's not really my fault.  a_http is the only default include without its own include guard.

**Comment by ziggi** [comment]
Fixed in 24f6860f049b0b8470193b502435d25aa33bc223
